src/rgw/driver/rados/rgw_sal_rados.cc

```diff
@@ -903,6 +910,169 @@ int RadosBucket::remove_topics(RGWObjVersionTracker* objv_tracker,
+                               data_pool,
+                               obj_name_oid,
+                               bl,
+                               nullptr,
+                               nullptr,
+                               y,
+                               dpp,
+                               nullptr,
+                               nullptr);
+  if (ret < 0) {
+    ldpp_dout(dpp, 1) << "failed to get logging object name from '" << obj_name_oid << "'. ret = " << ret << dendl;
+    return ret;
+  }
+  obj_name = bl.to_str();
+  return 0;
+}
+
+int RadosBucket::set_logging_object_name(const std::string& obj_name, const std::string& prefix, optional_yield y, const DoutPrefixProvider *dpp) {
+  rgw_pool data_pool;
+  const auto obj_name_oid = logging_object_name_oid(this, prefix);
+  if (!store->getRados()->get_obj_data_pool(get_placement_rule(), rgw_obj{get_key(), obj_name_oid}, &data_pool)) {
+    return -EIO;
+  }
+  bufferlist bl;
+  bl.append(obj_name); 
+  const int ret = rgw_put_system_obj(dpp, store->svc()->sysobj,
+                               data_pool,
+                               obj_name_oid,
+                               bl,
+                               false, // not exclusive - object may exist
+                               nullptr,
+                               ceph::real_time::clock::now(),
+                               y,
+                               nullptr);
+  if (ret < 0) {
+    ldpp_dout(dpp, 1) << "failed to set logging object name to '" << obj_name_oid << "'. ret = " << ret << dendl;
+    return ret;
+  }
+  return 0;
+}
+
+int RadosBucket::commit_logging_object(const std::string& obj_name, optional_yield y, const DoutPrefixProvider *dpp) {
+  rgw_pool data_pool;
+  const rgw_obj head_obj{get_key(), obj_name};
+  const auto placement_rule = get_placement_rule();
+
+  if (!store->getRados()->get_obj_data_pool(placement_rule, head_obj, &data_pool)) {
+    ldpp_dout(dpp, 1) << "failed to get data pool for bucket '" << get_name() << "' when comitting logging object. ret = " << -EIO << dendl;
+    return -EIO;
+  }
+
+  const auto temp_obj_name = to_temp_object_name(this, obj_name);
+  std::map<string, bufferlist> obj_attrs;
+  ceph::real_time mtime;
+  uint64_t size;
+  if (const auto ret = rgw_stat_system_obj(dpp,
+                        store->svc()->sysobj,
+                        data_pool,
+                        temp_obj_name,
+                        nullptr,
+                        &mtime,
+                        &size,
+                        y,
+                        &obj_attrs); ret < 0) {
+    ldpp_dout(dpp, 1) << "faild to read logging data when comitting to object '" << temp_obj_name
+      << ". error: " << ret << dendl;
+    return ret;
+  }
+
+
+  RGWObjManifest manifest;
+  manifest.set_prefix(obj_name);
+  manifest.set_trivial_rule(0, size);
+  RGWObjManifest::generator manifest_gen;
+  if (const auto ret = manifest_gen.create_begin(store->ctx(), &manifest,
```

Comment:
looks great. i think you're just missing a call to `manifest_gen.create_next(size)` after this (ex [AtomicObjectProcessor::complete()](https://github.com/ceph/ceph/blob/0fe3ccbd/src/rgw/driver/rados/rgw_putobj_processor.cc#L354-L355))

if this object write succeeds, you can use `radosgw-admin object stat --bucket=X --object=Y` to dump its manifest